jsonrpc/server.go

```diff
@@ -385,6 +390,10 @@ func (s *Server) buildArguments(ctx context.Context, params any, method Method)
 	case reflect.Slice:
 		paramsList := params.([]any)
 
+		if len(paramsList) == 0 && allParamsAreOptional {
+			return s.buildDefaultArguments(ctx, method)
+		}
+
```

Comment:
same, I dont think this is needed either. When using list of arguments (instead of a map) optional arguments are not supported in the spec I believe.

Reply:
Without this changes I get this error:
```
{
    "jsonrpc": "2.0",
    "error": {
        "code": -32602,
        "message": "Invalid Params",
        "data": "missing/unexpected params in list"
    },
    "id": 1
}
```

Reply:
Can you share the request and the handler definition?

Reply:
request is:
```
{
    "jsonrpc": "2.0",
    "id": 1,
    "method": "kirill_testing",
    "params": []
}
```
registration of this handler:
```
{
			Name:    "kirill_testing",
			Params:  []jsonrpc.Parameter{{Name: "v", Optional: true}},
			Handler: h.Testing,
		},
```
definition: 
```
func (h *Handler) Testing(v string) (string, *jsonrpc.Error) {
	return v, nil
}
```
I made a testing branch `kirugan/allow-nil-params-testing-handler` where this handler resides

Reply:
When passing the parameters as a list, parsing with optional parameters is not really straight forward. There is too much ambiguity. We can let it fail IMO.

Reply:
Ok, removed this check